src/group_theory/subgroup.lean

```diff
@@ -1375,3 +1376,101 @@ instance : is_modular_lattice (subgroup C) :=
 end⟩
 
 end subgroup
+
+section subgroup_monoid
+
+namespace subgroup
+
+/-- Two subgroups `H` and `K` can be multiplied to form another one. The result
+is the smallest subgroup that contains `{ h * k | h ∈ H, k ∈ K }`. -/
+instance subgroup_mul : has_mul (subgroup G) := ⟨λ H K, closure (H * K)⟩
```

Comment:
Isn't this precisely `H ⊔ K`?

Reply:
Oh yes, you're right! I guess then the only new contribution is the lemmas about normal subgroups. Is it worth keeping the monoid structure?

Reply:
If you do keep the monoid structure, it should be defined as defeq to sup. I also don't know if it's worth keeping though.

Can you add a lemma that says `a sup b = closure (a * b)`?

Reply:
Sure. I'll add that and keep the ones about normal subgroups.

Reply:
Done!